cmake/modules/FindZephyr-sdk.cmake

```diff
@@ -60,7 +60,44 @@ if(("zephyr" STREQUAL ${ZEPHYR_TOOLCHAIN_VARIANT}) OR
       set(ZEPHYR_TOOLCHAIN_VARIANT ${ZEPHYR_CURRENT_TOOLCHAIN_VARIANT})
     endif()
   else()
-    find_package(Zephyr-sdk ${Zephyr-sdk_FIND_VERSION} REQUIRED QUIET CONFIG PATHS
+    # Search for Zephyr SDK version 0.0.0 which does not exist, this is needed to
+    # return a list of compatible versions and find the best suited version that
+    # is available.
+    find_package(Zephyr-sdk 0.0.0 EXACT QUIET CONFIG PATHS
+                 /usr
+                 /usr/local
+                 /opt
+                 $ENV{HOME}
+                 $ENV{HOME}/.local
+                 $ENV{HOME}/.local/opt
+                 $ENV{HOME}/bin)
+
+    set(PACKAGE_FIND_VERSION ${Zephyr-sdk_FIND_VERSION})
+    set(COMPATIBLE_SDK_VERSION 0.0.0)
+
+    # Load the version file of each potential Zephyr SDK version and check if it
+    # is greater than the existing best found version.
+    foreach(ZEPHYR_SDK_CANDIDATE ${Zephyr-sdk_CONSIDERED_CONFIGS})
+      string(REGEX REPLACE "\.cmake$" "Version.cmake" ZEPHYR_SDK_VERSION_CANDIDATE ${ZEPHYR_SDK_CANDIDATE})
+
+      include(${ZEPHYR_SDK_VERSION_CANDIDATE} NO_POLICY_SCOPE)
+
+      if("${PACKAGE_VERSION_COMPATIBLE}" STREQUAL "TRUE")
+        if("${PACKAGE_VERSION}" VERSION_GREATER "${COMPATIBLE_SDK_VERSION}")
+          set(COMPATIBLE_SDK_VERSION ${PACKAGE_VERSION})
+        endif()
+      endif()
+    endforeach()
+
+    if("${COMPATIBLE_SDK_VERSION}" STREQUAL "0.0.0")
+      # This means no compatible Zephyr SDK versions were found, set the version
+      # back to the minimum version so that it is displayed in the error text.
+      set(COMPATIBLE_SDK_VERSION ${Zephyr-sdk_FIND_VERSION})
+    endif()
+
+    # Now that the best version has been found, find the exact Zephyr SDK of this
+    # version and load it.
+    find_package(Zephyr-sdk ${COMPATIBLE_SDK_VERSION} EXACT REQUIRED QUIET CONFIG PATHS
```

Comment:
why not simply order the `Zephyr-sdk_CONSIDERED_CONFIGS` list from newest to oldest, and then call this `find_package(<version-to-try> EXACT)` with newest first, until `PACKAGE_VERSION_COMPATIBLE` is true ?

Reply:
Fixed